Improvements to R2R Test Harness

nattress opened the pull request:
To support multi-core testing, associate the test process's PID with the ETW filter so that only assemblies loaded by the runtime the harness started are considered.

Emit the assembly name for methods that are jitted to allow grouping of methods by their assembly.

Add a new parameter, `--include` which configures the ETW filter to include all assemblies in a given folder, instead of having to specify a large number of assemblies individually.

Bug fixes around path casing causing events to be dropped.

trylek commented on tests/src/tools/ReadyToRun.TestHarness/PathHelpers.cs:
```diff
@@ -0,0 +1,34 @@
+    /// A set of helper to manipulate paths into a canonicalized form to ensure user-provided paths
+    /// match those in the ETW log.
+    /// </summary>
+    static class PathHelpers
```
Nit - I was under the impressions that we typically call such classes "###Extensions", not 100% sure how much of a code style requirement that is.

trylek commented:
By the way, it happens to me quite often in local CPAOT testing that I receive the same error message ("No assemblies have been loaded") even though the behavior is somewhat different - when I get the message, the tests don't run, while in the lab it looks like the test first successfully ran to completion and only then something failed in the harness, presumably some code related to the ETW analysis.

The difference may be caused by your behavioral changes to the tool that are part of this PR. One way or another, when this happens to me, I usually overwrite the CoreClrRuntime folder with the Core_Root folder from my local CoreCLR build, that fixes the issue; but it's somewhat annoying and I assume that, if the CPAOT project gains momentum and more people start hacking around the code, we'll hear complaints more often.

trylek commented on tests/src/tools/ReadyToRun.TestHarness/Program.cs:
```diff
@@ -203,7 +222,12 @@ private static int AnalyzeResults(ReadyToRunJittedMethods jittedMethods, string
-        private static void RunTest(TraceEventSession session, string testArguments, out int exitCode)
+        private static string GetFullModuleName(string moduleName, string methodName)
+        {
+            return $"{moduleName}!{methodName}";
```
Nit - while the exclamation mark is a natural choice for a WinDbg user, we might want to spend a minute thinking whether this cannot cause trouble for generic method names which may contain exclamation marks in their signatures to indicate the presence of generic arguments.

nattress commented:
Do you mean you were seeing "No assemblies have been loaded" before my PR?  Or you were using these changes locally?  I fixed an issue with casing that was causing no assemblies loaded errors if I wasn't careful with the command line input. I fixed that but didn't lower the case on the input exe for the simple tests. I fixed that in a follow-up commit.

trylek commented:
I used to see "No assemblies have been loaded" before your PR. It's great if it's something you managed to fix as part of this change.
